src/components/content/rich-text/rich-text.tsx

```diff
@@ -0,0 +1,330 @@
+import { Options } from '@contentful/rich-text-react-renderer';
+import { BLOCKS, INLINES, MARKS } from '@contentful/rich-text-types';
+import { GatsbyImage } from 'gatsby-plugin-image';
+import {
+  ContentfulRichTextGatsbyReference,
+  renderRichText,
+  RenderRichTextData,
+} from 'gatsby-source-contentful/rich-text';
+import React, { MouseEvent } from 'react';
+import { scrollToTarget } from '../../layout/use-anchor-tag-scrolling';
+import customComponents from '../../legacy/markdown/custom-components';
+import { MarkdownAst } from '../../legacy/markdown/markdown-ast';
+import Zoom from 'react-medium-image-zoom';
+import 'react-medium-image-zoom/dist/styles.css';
+import { StatItem, Stats } from '../../content/stats/stats';
+import { AdvancedAsset } from '../../content/advanced-asset/advanced-asset';
+
+interface RichTextProps {
+  data: RenderRichTextData<ContentfulRichTextGatsbyReference>;
+}
+
+interface ContentfulFootnoteReference
+  extends ContentfulRichTextGatsbyReference {
+  note: any; // TODO: Improve typing here
+}
+
+enum ContentfulCustomModel {
+  CONTENTFUL_FOOTNOTE = 'ContentfulFootnote',
+  CONTENTFUL_CODE_BLOCK = 'ContentfulCodeBlock',
+  CONTENTFUL_BLOG_POST_COLLAPSIBLE = 'ContentfulBlogPostCollapsible',
+  CONTENTFUL_STATS = 'ContentfulStats',
+  CONTENTFUL_ADVANCED_ASSET = 'ContentfulAdvancedAsset',
+}
+
+enum EmbeddedAssetType {
+  IMAGE = 'image',
+  VIDEO = 'video',
+}
+interface FootnoteReference {
+  [key: string]: {
+    index: number;
+    anchor: string;
+    referenceAnchor: string;
+  };
+}
+
+/**
+ * Custom Contentful Rich Text Renderer
+ *
+ * Will automatically filter out ContentfulFootnote elements and appends an
+ * ordered list of footnotes after the rendered rich text.
+ *
+ * @param data rich text data
+ * @returns rendered elements
+ */
+const customContentfulRenderer = (
```

Comment:
This can also have a better name, like `ContentfulRichTextRenderer`, same with the file name. The rich text is explicitly for Contentful rich text, so `contentful` should also be in the file name.